great_expectations/rule_based_profiler/builder.py

```diff
@@ -39,9 +39,9 @@ def __init__(
         Args:
             data_context: AbstractDataContext associated with this Builder
         """
-        self._batch_list = None
-        self._batch_request = None
-        self._data_context = data_context
+        self._batch_list: List[Batch] = []
```

Comment:
This is one of the only runtime changes in this PR.

Reply:
I'm going to have to revert this change.
There's at least one `if batch_list is None:` style check that makes this not work.

To make this work we would need to change these to `if not batch_list:`.